### ISSUE TITLE
I/flutter ( 9033): Another exception was thrown: 'package:flutter/src/rendering/sliver.dart': Failed assertion: line 556 pos 15: 'scrollOffsetCorrection != 0.0': is not true. I/flutter ( 9033): Another exception was thrown: NoSuchMethodError: The method '-' was called on null.

### ISSUE THREAD

**Issue by MubashirMaqbool** [flutter/flutter#63263]
```
I/flutter ( 9033): The following RenderObject was being processed when the exception was fired: RenderSliverList#3776a relayoutBoundary=up2 NEEDS-LAYOUT NEEDS-COMPOSITING-BITS-UPDATE:
I/flutter ( 9033):   needs compositing
I/flutter ( 9033):   creator: SliverList ← MediaQuery ← SliverPadding ← Viewport ← IgnorePointer-[GlobalKey#997ae] ←
I/flutter ( 9033):     Semantics ← _PointerListener ← Listener ← _GestureSemantics ←
I/flutter ( 9033):     RawGestureDetector-[LabeledGlobalKey<RawGestureDetectorState>#6878b] ← _PointerListener ← Listener
I/flutter ( 9033):     ← ⋯
I/flutter ( 9033):   parentData: paintOffset=Offset(0.0, 0.0) (can use size)
I/flutter ( 9033):   constraints: SliverConstraints(AxisDirection.down, GrowthDirection.forward, ScrollDirection.forward,
I/flutter ( 9033):     scrollOffset: 174.1, remainingPaintExtent: 566.0, crossAxisExtent: 360.0, crossAxisDirection:
I/flutter ( 9033):     AxisDirection.right, viewportMainAxisExtent: 566.0, remainingCacheExtent: 990.1, cacheOrigin:
I/flutter ( 9033):     -174.1)
I/flutter ( 9033):   geometry: SliverGeometry(scrollExtent: 2552.0, paintExtent: 566.0, maxPaintExtent: 2552.0,
I/flutter ( 9033):     hasVisualOverflow: true, cacheExtent: 1066.0)
I/flutter ( 9033):   currently live children: 0 to 1
I/flutter ( 9033): This RenderObject had the following descendants (showing up to depth 5):
I/flutter ( 9033):     child with index 0: RenderIndexedSemantics#d8a49 relayoutBoundary=up3 NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
I/flutter ( 9033):       child: RenderRepaintBoundary#c4a34 relayoutBoundary=up4 NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
I/flutter ( 9033):         child: RenderLimitedBox#1d176 relayoutBoundary=up5 NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
I/flutter ( 9033):           child: RenderConstrainedBox#9aaa2 NEEDS-PAINT
I/flutter ( 9033):     child with index 1: RenderIndexedSemantics#31bed relayoutBoundary=up3
I/flutter ( 9033):       child: RenderRepaintBoundary#d9ec1 relayoutBoundary=up4
I/flutter ( 9033):         child: RenderFlex#173fc relayoutBoundary=up5
I/flutter ( 9033):           child 1: RenderSemanticsAnnotations#6823c relayoutBoundary=up6
I/flutter ( 9033):             child: RenderImage#612e2 relayoutBoundary=up7
I/flutter ( 9033):           child 2: RenderSemanticsAnnotations#ad8d0 relayoutBoundary=up6
I/flutter ( 9033):             child: RenderImage#20b03 relayoutBoundary=up7
I/flutter ( 9033):           child 3: RenderSemanticsAnnotations#384ce relayoutBoundary=up6
I/flutter ( 9033):             child: RenderImage#9aa77 relayoutBoundary=up7
I/flutter ( 9033):           child 4: RenderSemanticsAnnotations#1af8a relayoutBoundary=up6
I/flutter ( 9033):             child: RenderImage#c08c6 relayoutBoundary=up7
I/flutter ( 9033):           child 5: RenderSemanticsAnnotations#7340a relayoutBoundary=up6
I/flutter ( 9033):             child: RenderImage#26937 relayoutBoundary=up7
I/flutter ( 9033):           child 6: RenderSemanticsAnnotations#3def5 relayoutBoundary=up6
I/flutter ( 9033):             child: RenderImage#4c1cf relayoutBoundary=up7
I/flutter ( 9033):           child 7: RenderSemanticsAnnotations#c6d29 relayoutBoundary=up6
I/flutter ( 9033):             child: RenderImage#edb43 relayoutBoundary=up7
I/flutter ( 9033):           child 8: RenderSemanticsAnnotations#9c222 relayoutBoundary=up6
I/flutter ( 9033):             child: RenderImage#d0491 relayoutBoundary=up7
I/flutter ( 9033): ════════════════════════════════════════════════════════════════════════════════════════════════════
I/flutter ( 9033): Another exception was thrown: NoSuchMethodError: The method '-' was called on null.
I/flutter ( 9033): Another exception was thrown: 'package:flutter/src/rendering/sliver.dart': Failed assertion: line 556 pos 15: 'scrollOffsetCorrection != 0.0': is not true.
I/flutter ( 9033): Another exception was thrown: NoSuchMethodError: The method '-' was called on null.

```

**Comment by roohaani786** [comment]
@MubashirMaqbool  Can you share me the code, so that i can take a peek. I think i may have encountered and resolved this.

**Comment by praveengupta0895** [comment]
I'm also facing the same problem after upgrading flutter to newer version.

**Comment by MubashirMaqbool** [comment]
@roohaani786 
@override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(backgroundColor: color.Red,title: Text(widget.Name.toUpperCase(),style: TextStyle(color: Colors.white),),),body:
    ListView.builder(itemBuilder: (context,index){

     return _games[index].id==widget.CatId?Column(
children: [



for(int i=0;i<_games[index]._Gamename.length;i++)
  Card(
    elevation: 5,
    shape: RoundedRectangleBorder(
      borderRadius: BorderRadius.circular(15.0),),
    clipBehavior: Clip.antiAlias,

    child: Stack(children: [
Center(child: Text('loading......')),
     InkWell(
         onTap: (){
           Navigator.push(
             context,
             MaterialPageRoute(builder: (context) => GameDetail(


                 Name:_games[index]._Gamename[i],
                 Image:_games[index]._GameUrl[i],
                 Des:_games[index]._Description[i],
                 Target:_games[index]._TargetArea[i]

             )),
           );

         },
         child: Container(height: SizeConfig.imageSizeMultiplier*30,decoration: BoxDecoration(image: DecorationImage(image: AssetImage(_games[index]._GameUrl[i]),fit: BoxFit.cover)),child: Container(color: Colors.black26,),)),
      Positioned(
     top: 25,
          left: 15,
          child: Text(_games[index]._Gamename[i],style: TextStyle(color: Colors.white,fontWeight: FontWeight.bold),)),


    ],),
  )
],
     ):Container();

    },itemCount: _games.length,)
    );
  }

**Comment by Mudassar823** [comment]
i am also facing same issue

**Comment by Mudassar823** [comment]
my code is 
body:
    ListView.builder(itemBuilder: (context,index){

      return _games[index].id==widget.CatId?Column(
        children: [




            Card(
              elevation: 5,
              shape: RoundedRectangleBorder(
                borderRadius: BorderRadius.circular(15.0),),
              clipBehavior: Clip.antiAlias,

              child: Stack(children: [
                Center(child: Text('loading......')),
                InkWell(
                    onTap: (){
                      Navigator.push(
                        context,
                        MaterialPageRoute(builder: (context) => GameDetail(


                            Name:_games[index]._Gamename,
                            Image:_games[index]._GameUrl,
                            Des:_games[index]._Description,
                            Target:_games[index]._TargetArea

                        )),
                      );

                    },
                    child: Container(height: SizeConfig.imageSizeMultiplier*30,decoration: BoxDecoration(image: DecorationImage(image: AssetImage(_games[index]._GameUrl),fit: BoxFit.cover)),child: Container(color: Colors.black26,),)),
                Positioned(
                    top: 25,
                    left: 15,
                    child: Text(_games[index]._Gamename,style: TextStyle(color: Colors.white,fontWeight: FontWeight.bold),)),


              ],),
            )
        ],
      ):Container();

    },itemCount: _games.length,)
    );

**Comment by Mudassar823** [comment]
@praveengupta0895 have you resolved the issue?

**Comment by darshankawar** [comment]
Hi @MubashirMaqbool,
This platform is not meant for assistance on personal code. Please see https://flutter.dev/community for resources and asking questions like this.
You may also get some help if you post it on Stack Overflow and if you need help with your code, please see https://www.reddit.com/r/flutterhelp/
Closing, as this isn't an issue with Flutter itself. If you disagree, please write in the comments and I will reopen it.
Thank you

**Comment by praveengupta0895** [comment]
Hi @darshankawar , I understand your concern but as far as I think this is some kind of flutter issue because earlier the code was working fine and with the recent update of flutter we are facing this problem.

**Comment by praveengupta0895** [comment]
Hello @Mudassar823, still not able to resolve it, and right now I'm downgrading my flutter. because I've the same application with different name and older version of flutter and it is running perfect.
And the problem we are facing is due to this line "_games[index].id==widget.CatId". If you remove the condition it will work.

**Comment by Mudassar823** [comment]
@praveengupta0895  I have to show data on basis of id, this is not a proper solution brother

**Comment by Mudassar823** [comment]
@praveengupta0895  I am facing this issue
I/flutter ( 9657): Another exception was thrown: 'package:flutter/src/rendering/sliver.dart': Failed assertion: line 556 pos 15: 'scrollOffsetCorrection != 0.0': is not true.
I/flutter ( 9657): Another exception was thrown: NoSuchMethodError: The method '-' was called on null.

**Comment by praveengupta0895** [comment]
No, I was not asking you to remove that ID. I was saying that error is showing due to that only.

**Comment by Mudassar823** [comment]
@praveengupta0895 ok brother we have to wait and be patient

**Comment by Mudassar823** [comment]
@praveengupta0895 have you asked it on stackoverflow

**Comment by praveengupta0895** [comment]
I think as of now until flutter community fix this issue. we only have 2 options either downgrade it to 1.19 version or grouped the data.

**Comment by praveengupta0895** [comment]
Yes I've asked it on stack overflow, and also raised another issue on github, but didn't got the proper answer.

**Comment by praveengupta0895** [comment]
Once go through with this link  "https://pub.dev/packages/grouped_list" This might help you.

**Comment by Mudassar823** [comment]
@praveengupta0895 ok

**Comment by praveengupta0895** [comment]
I just saw your code, I think I can help you with your code as you are comparing the data with the single item.

**Comment by Mudassar823** [comment]
@praveengupta0895 I am passing data from one page with id and then compare this data with next page id's?
how can you help me

**Comment by praveengupta0895** [comment]
define another list _game2 and add data to _game2 after comparing.

        _games.add(data);
        _games2 = _games.where((i) => i.ID ==widget.ID).toList();

**Comment by Mudassar823** [comment]
@praveengupta0895  everything is correct issue in scrolling

**Comment by praveengupta0895** [comment]
If you can share the whole code then I might be able to fix it.

**Comment by praveengupta0895** [comment]
and also share the error.

**Comment by Mudassar823** [comment]
List<GamesInfo> _games=[
    GamesInfo('Dumbbell Pullover','assets/chest/Dumble pull over.JPG',1,'pectorals, latissimus dorsi, serratus','Lie on a bench with your upper back, head and neck supported and your feet flat on the floor. Hold a dumbbell with your arms extended above your face. Maintaining just a slight bend in your elbow throughout, slowly lower the dumbbell backward, allowing your elbows to come to a point at which they align with your ears. When you’ve stretched as far as you can without bending your elbows, flex through your chest and lats to reverse direction to bring the dumbbell back overhead.'),
    GamesInfo('Pec-Deck Flye','assets/chest/butterfly.JPG',1,'middle pecs','Sit with your lower back fully in contact with the pad and your feet flat on the floor. With your elbows at 90-degree angles and forearms flush against the pads, move your arms slightly forward to disengage the weight from the stack. From here, deliberately flex your pecs to bring the handles out in an arc until they meet in front of your body. Squeeze hard, then slowly return to the start, stopping when your upper arms are even with your torso, and repeat.'),
    GamesInfo('Decline Bench Press','assets/chest/decline bench press.JPG',1,'inner pecs, lower pecs','Lie on a decline bench. Your torso should be fully supported from your head to your hips, with your knees bent and feet harnessed behind roller pads. Grasp the bar with a wide, overhand grip. Bend your arms and slowly lower the bar toward your lower chest. When the bar reaches chest level, forcefully extend your arms, elevating the bar back to the starting position.'),
    GamesInfo('Cable Crossover','assets/chest/cablecross.png',1,'middle pecs','Stand in the direct center of a cable-cross station with your feet staggered, knees slightly bent and your focus forward, and grasp D-handles attached to the upper pulleys. Starting with your palms facing downward and elbows bent slightly, flex your pecs to draw the handles down and together, meeting below your waist. Try to keep your elbows up throughout the movement. Pause a moment for a peak contraction, then slowly allow the handles to return to the start position. Don’t let the weight stacks touch down between reps.'),
    GamesInfo('Dumbbell Bench Press','assets/chest/dumble press.jpg',1,'upper pecs','Lie on a bench with your feet flat on the floor, holding a dumbbell in each hand just outside your shoulders. Powerfully press the dumbbells upward toward the ceiling, stopping when they come to an inch or so away from each other above your upper-middle chest, then slowly bend your elbows to lower the weights back down to a point even with your torso.'),
    GamesInfo('Incline Dumbbell Flye','assets/chest/incline duble fly.JPG',1,'upper pecs','middle pecs'),
    GamesInfo('Standing Cable Curl','assets/biceps/cable curl.JPG',2,'Biceps brachii from origin to insertion and peak','Stand holding a bar attached to a low-pulley cable with an underhand grip, elbows extended. Keep your abs tight, chest up and head straight as you contract your biceps to curl the bar toward your chest, keeping your elbows at your sides throughout. Hold and squeeze the contraction at the top, then slowly return the bar along the same path. Repeat for reps without letting the weight stack touch down between reps.'),

    GamesInfo('Seated Alternating Dumbbell Curl','assets/biceps/dumble curl.JPG',2,'Biceps brachii, with brachialis activation if using a rope','Stand inside a Smith machine holding the bar in front of your upper thighs, with your chest up, shoulders back and eyes focused straight ahead. Begin by shifting your elbows back and bending them to curl the bar toward your upper abs. As the name suggests, you’ll actually drag the bar up your torso, keeping your elbows behind you the entire time. Upon full contraction, return the bar along the same path and repeat'),

    GamesInfo('Dumbbell Concentration Curl','assets/biceps/dumble concentration curl.JPG',2,'Biceps and brachialis, emphasizing the full length of the muscle','Sit on a low-back bench, holding a dumbbell in each hand at your sides. Keeping your chest up, curl one weight toward the same-side shoulder, squeezing your biceps hard at the top, then lower to the start. Repeat with the other arm. You can either condense the set by lowering one dumbbell as you lift the other, or take it slower by moving through one full rep per side, going all the way up and down with one arm before switching to the other.'),

    GamesInfo('Scott Curl','assets/biceps/scott curl.JPG',2,'Biceps, with emphasis on the peak','Sit at the end of a flat bench or use a short-back chair. Bend over and grasp a dumbbell with an underhand grip, locking your working arm against your same-side inner thigh. Place your nonworking hand on your leg for balance. Moving only at your elbow, curl the weight as high as you can toward your torso. Squeeze your biceps at the top before lowering the dumbbell back to the start, but don’t let it rest on the floor between reps.'),
    GamesInfo('Barbell Preacher Curl','assets/biceps/preacher curl.JPG',2,'Biceps, with emphasis on the short, inner head','Grasp a barbell or EZ-bar and lean your chest against the angled side of a preacher bench, keeping your back tight and knees slightly bent. (You can also use a dumbbell and train one arm at a time.) Make sure your armpits fit securely against the top of the pad, with your triceps pressed into the flat side of the bench. Hold the bar straight toward the floor with a supinated (palms-up) grip. With your head neutral and eyes focused forward, curl the weight in a smooth arc, squeeze your biceps hard for a one-count, and slowly lower back to the nearly elbows-straight position.'),
    GamesInfo('Incline Dumbbell Curl','assets/biceps/incline dumble curl.JPG',2,'Biceps, with emphasis on the stretch','Set up a preacher bench so the top of the pad fits securely under your armpits. Take a shoulder-width, underhand grip on a straight bar or EZ-bar and position your arms parallel to each other on the pad. Keep your feet flat on the floor and your head straight as you flex your biceps to bring the bar as high as possible. Don’t allow your elbows to flare out or come off the bench. Flex strongly at the top before slowly returning the bar along the same path, stopping just short of full elbow extension before starting the next rep. You can also do this exercise one arm at a time with a dumbbell.'),
    GamesInfo('Close-Grip Chin-Up','assets/biceps/close grip chin up.JPG',2,'Biceps and brachialis, emphasizing the full length of the muscle','Adjust an incline bench to about 45 degrees and sit back squarely against the pad, feet flat on the floor. Your arms should hang straight down by your sides, palms up. With your shoulders shifted back and upper arms in a fixed position perpendicular to the floor, curl the weights so they approach your shoulders, either one at a time or both arms simultaneously. Squeeze your biceps hard at the top before slowly returning to the start position.'),
    GamesInfo('Barbell Curl','assets/biceps/barbel curl.JPG',2,'Brachialis mainly, with biceps brachii activation ','Grasp a fixed overhead bar with a close underhand grip. Hang freely from the bar, arms fully extended and ankles crossed behind you. Contract your lats and bi’s to bring your body upward, concentrating on keeping your elbows tight to your sides. Hold momentarily as your chin crosses the level of the bar, then lower yourself to return to the dead-hang position'),
  ];
  @override
  void initState() {
    // TODO: implement initState
    super.initState();

  }
  @override
  Widget build(BuildContext context) {
    return Scaffold(
        appBar: AppBar(backgroundColor: color.Red,title: Text(widget.Name.toUpperCase(),style: TextStyle(color: Colors.white),),),body:
    ListView.builder(
      scrollDirection: Axis.horizontal,
      itemBuilder: (context,index){

      return _games[index].id==widget.CatId?Column(
        children: [
            Card(
              elevation: 5,
              shape: RoundedRectangleBorder(
                borderRadius: BorderRadius.circular(15.0),),
              clipBehavior: Clip.antiAlias,

              child: Stack(children: [
                Center(child: Text('loading......')),
                InkWell(
                    onTap: (){
                      Navigator.push(
                        context,
                        MaterialPageRoute(builder: (context) => GameDetail(


                            Name:_games[index]._Gamename,
                            Image:_games[index]._GameUrl,
                            Des:_games[index]._Description,
                            Target:_games[index]._TargetArea

                        )),
                      );

                    },
                    child: Container(height: SizeConfig.imageSizeMultiplier*30,decoration: BoxDecoration(image: DecorationImage(image: AssetImage(_games[index]._GameUrl),fit: BoxFit.cover)),child: Container(color: Colors.black26,),)),
                Positioned(
                    top: 25,
                    left: 15,
                    child: Text(_games[index]._Gamename,style: TextStyle(color: Colors.white,fontWeight: FontWeight.bold),)),


              ],),
            )
        ],
      ):Container();

    },itemCount: _games.length,)
    );
  }


}

**Comment by Mudassar823** [comment]
List<GamesInfo> _games=[
    GamesInfo('Dumbbell Pullover','assets/chest/Dumble pull over.JPG',1,'pectorals, latissimus dorsi, serratus','Lie on a bench with your upper back, head and neck supported and your feet flat on the floor. Hold a dumbbell with your arms extended above your face. Maintaining just a slight bend in your elbow throughout, slowly lower the dumbbell backward, allowing your elbows to come to a point at which they align with your ears. When you’ve stretched as far as you can without bending your elbows, flex through your chest and lats to reverse direction to bring the dumbbell back overhead.'),
    GamesInfo('Pec-Deck Flye','assets/chest/butterfly.JPG',1,'middle pecs','Sit with your lower back fully in contact with the pad and your feet flat on the floor. With your elbows at 90-degree angles and forearms flush against the pads, move your arms slightly forward to disengage the weight from the stack. From here, deliberately flex your pecs to bring the handles out in an arc until they meet in front of your body. Squeeze hard, then slowly return to the start, stopping when your upper arms are even with your torso, and repeat.'),
    GamesInfo('Decline Bench Press','assets/chest/decline bench press.JPG',1,'inner pecs, lower pecs','Lie on a decline bench. Your torso should be fully supported from your head to your hips, with your knees bent and feet harnessed behind roller pads. Grasp the bar with a wide, overhand grip. Bend your arms and slowly lower the bar toward your lower chest. When the bar reaches chest level, forcefully extend your arms, elevating the bar back to the starting position.'),
    GamesInfo('Cable Crossover','assets/chest/cablecross.png',1,'middle pecs','Stand in the direct center of a cable-cross station with your feet staggered, knees slightly bent and your focus forward, and grasp D-handles attached to the upper pulleys. Starting with your palms facing downward and elbows bent slightly, flex your pecs to draw the handles down and together, meeting below your waist. Try to keep your elbows up throughout the movement. Pause a moment for a peak contraction, then slowly allow the handles to return to the start position. Don’t let the weight stacks touch down between reps.'),
    GamesInfo('Dumbbell Bench Press','assets/chest/dumble press.jpg',1,'upper pecs','Lie on a bench with your feet flat on the floor, holding a dumbbell in each hand just outside your shoulders. Powerfully press the dumbbells upward toward the ceiling, stopping when they come to an inch or so away from each other above your upper-middle chest, then slowly bend your elbows to lower the weights back down to a point even with your torso.'),
    GamesInfo('Incline Dumbbell Flye','assets/chest/incline duble fly.JPG',1,'upper pecs','middle pecs'),
    GamesInfo('Standing Cable Curl','assets/biceps/cable curl.JPG',2,'Biceps brachii from origin to insertion and peak','Stand holding a bar attached to a low-pulley cable with an underhand grip, elbows extended. Keep your abs tight, chest up and head straight as you contract your biceps to curl the bar toward your chest, keeping your elbows at your sides throughout. Hold and squeeze the contraction at the top, then slowly return the bar along the same path. Repeat for reps without letting the weight stack touch down between reps.'),

    GamesInfo('Seated Alternating Dumbbell Curl','assets/biceps/dumble curl.JPG',2,'Biceps brachii, with brachialis activation if using a rope','Stand inside a Smith machine holding the bar in front of your upper thighs, with your chest up, shoulders back and eyes focused straight ahead. Begin by shifting your elbows back and bending them to curl the bar toward your upper abs. As the name suggests, you’ll actually drag the bar up your torso, keeping your elbows behind you the entire time. Upon full contraction, return the bar along the same path and repeat'),

    GamesInfo('Dumbbell Concentration Curl','assets/biceps/dumble concentration curl.JPG',2,'Biceps and brachialis, emphasizing the full length of the muscle','Sit on a low-back bench, holding a dumbbell in each hand at your sides. Keeping your chest up, curl one weight toward the same-side shoulder, squeezing your biceps hard at the top, then lower to the start. Repeat with the other arm. You can either condense the set by lowering one dumbbell as you lift the other, or take it slower by moving through one full rep per side, going all the way up and down with one arm before switching to the other.'),

    GamesInfo('Scott Curl','assets/biceps/scott curl.JPG',2,'Biceps, with emphasis on the peak','Sit at the end of a flat bench or use a short-back chair. Bend over and grasp a dumbbell with an underhand grip, locking your working arm against your same-side inner thigh. Place your nonworking hand on your leg for balance. Moving only at your elbow, curl the weight as high as you can toward your torso. Squeeze your biceps at the top before lowering the dumbbell back to the start, but don’t let it rest on the floor between reps.'),
    GamesInfo('Barbell Preacher Curl','assets/biceps/preacher curl.JPG',2,'Biceps, with emphasis on the short, inner head','Grasp a barbell or EZ-bar and lean your chest against the angled side of a preacher bench, keeping your back tight and knees slightly bent. (You can also use a dumbbell and train one arm at a time.) Make sure your armpits fit securely against the top of the pad, with your triceps pressed into the flat side of the bench. Hold the bar straight toward the floor with a supinated (palms-up) grip. With your head neutral and eyes focused forward, curl the weight in a smooth arc, squeeze your biceps hard for a one-count, and slowly lower back to the nearly elbows-straight position.'),
    GamesInfo('Incline Dumbbell Curl','assets/biceps/incline dumble curl.JPG',2,'Biceps, with emphasis on the stretch','Set up a preacher bench so the top of the pad fits securely under your armpits. Take a shoulder-width, underhand grip on a straight bar or EZ-bar and position your arms parallel to each other on the pad. Keep your feet flat on the floor and your head straight as you flex your biceps to bring the bar as high as possible. Don’t allow your elbows to flare out or come off the bench. Flex strongly at the top before slowly returning the bar along the same path, stopping just short of full elbow extension before starting the next rep. You can also do this exercise one arm at a time with a dumbbell.'),
    GamesInfo('Close-Grip Chin-Up','assets/biceps/close grip chin up.JPG',2,'Biceps and brachialis, emphasizing the full length of the muscle','Adjust an incline bench to about 45 degrees and sit back squarely against the pad, feet flat on the floor. Your arms should hang straight down by your sides, palms up. With your shoulders shifted back and upper arms in a fixed position perpendicular to the floor, curl the weights so they approach your shoulders, either one at a time or both arms simultaneously. Squeeze your biceps hard at the top before slowly returning to the start position.'),
    GamesInfo('Barbell Curl','assets/biceps/barbel curl.JPG',2,'Brachialis mainly, with biceps brachii activation ','Grasp a fixed overhead bar with a close underhand grip. Hang freely from the bar, arms fully extended and ankles crossed behind you. Contract your lats and bi’s to bring your body upward, concentrating on keeping your elbows tight to your sides. Hold momentarily as your chin crosses the level of the bar, then lower yourself to return to the dead-hang position'),
  ];
  @override
  void initState() {
    // TODO: implement initState
    super.initState();

  }
  @override
  Widget build(BuildContext context) {
    return Scaffold(
        appBar: AppBar(backgroundColor: color.Red,title: Text(widget.Name.toUpperCase(),style: TextStyle(color: Colors.white),),),body:
    ListView.builder(
      scrollDirection: Axis.horizontal,
      itemBuilder: (context,index){

      return _games[index].id==widget.CatId?Column(
        children: [
            Card(
              elevation: 5,
              shape: RoundedRectangleBorder(
                borderRadius: BorderRadius.circular(15.0),),
              clipBehavior: Clip.antiAlias,

              child: Stack(children: [
                Center(child: Text('loading......')),
                InkWell(
                    onTap: (){
                      Navigator.push(
                        context,
                        MaterialPageRoute(builder: (context) => GameDetail(


                            Name:_games[index]._Gamename,
                            Image:_games[index]._GameUrl,
                            Des:_games[index]._Description,
                            Target:_games[index]._TargetArea

                        )),
                      );

                    },
                    child: Container(height: SizeConfig.imageSizeMultiplier*30,decoration: BoxDecoration(image: DecorationImage(image: AssetImage(_games[index]._GameUrl),fit: BoxFit.cover)),child: Container(color: Colors.black26,),)),
                Positioned(
                    top: 25,
                    left: 15,
                    child: Text(_games[index]._Gamename,style: TextStyle(color: Colors.white,fontWeight: FontWeight.bold),)),


              ],),
            )
        ],
      ):Container();

    },itemCount: _games.length,)
    );
  }


}

**Comment by Mudassar823** [comment]
axis not horizental its vertical

**Comment by praveengupta0895** [comment]
Okay so first of all define another list of same model. and apply the above comparison code, remove comparison while returning Column and to avoid scrolling problem wrap ListView.builder into Expanded widget.

**Comment by Mudassar823** [comment]
@praveengupta0895 okay I check

**Comment by darshankawar** [comment]
@praveengupta0895  @Mudassar823 please follow-up your discussion privately.
